.github/workflows/publish-pypi.yml

```diff
@@ -0,0 +1,67 @@
+---
+# yamllint disable rule:truthy
+name: Publish Infrahub-Server Package
+
+on:
+  workflow_call:
+    inputs:
+      runs-on:
+        description: "The OS to run the job on"
+        required: false
+        default: "ubuntu-22.04"
+        type: string
+      version:
+        type: string
+        required: false
+        description: The string to extract semver labels from.
+        default: ''
+      publish:
+        type: boolean
+        description: Wether to publish the package to Pypi
+        required: false
+        default: false
+
+jobs:
+  publish_to_pypi:
+    name: "Publish Infrahub SDK to PyPI"
+    runs-on: "ubuntu-22.04"
+    steps:
+      - name: "Set up Python"
+        uses: "actions/setup-python@v5"
+        with:
+          python-version: "3.12"
+
+      - name: "Install Poetry"
+        uses: "snok/install-poetry@v1"
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+
+      - name: "Check out repository code"
+        uses: "actions/checkout@v4"
+
+      - name: "Cache poetry venv"
+        uses: "actions/cache@v4"
+        id: "cached-poetry-dependencies"
+        with:
+          path: "./infrahub_sync/.venv"
```

Comment:
I'm guessing this is copied from another repo.

Reply:
spotted on Patrick :p 
It was a kickstart for @fatih-acar  to continue but yeah bad copy-paste